command/meta.go

```diff
@@ -63,13 +66,19 @@ type Meta struct {
 func (m *Meta) FlagSet(n string, fs FlagSetFlags) *flag.FlagSet {
 	f := flag.NewFlagSet(n, flag.ContinueOnError)
 
+	forceColor := false
+	if os.Getenv(EnvNomadCLIForceColor) != "" {
+		forceColor = true
+	}
```

Comment:
This seems like it is being used to set the default value on the flag, whereas the default value should be fixed and not mutable by env vars. I think we can remove this line and update the default flag value to be false.

Reply:
Simply changing `meta.forceColor` to `*bool` and fixing up the code so it compiles crashes at runtime in `FlagSet`:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1f08e75]

goroutine 1 [running]:
flag.newBoolValue(...)
	flag/flag.go:112
flag.(*FlagSet).BoolVar(...)
	flag/flag.go:630
github.com/hashicorp/nomad/command.(*Meta).FlagSet(0xc00078da00, 0x2862ddc, 0x8, 0x2, 0x4c0b58)
	github.com/hashicorp/nomad/command/meta.go:75 +0x1d5
github.com/hashicorp/nomad/command.(*JobPlanCommand).Run(0xc00078da00, 0xc0001b2030, 0x1, 0x1, 0xc0003465c0)
	github.com/hashicorp/nomad/command/job_plan.go:122 +0xf1
github.com/mitchellh/cli.(*CLI).Run(0xc00034ea00, 0xc00034ea00, 0xc0001a0300, 0x43)
	github.com/mitchellh/cli@v1.1.0/cli.go:260 +0x41a
main.RunCustom(0xc0001b2010, 0x3, 0x3, 0xc000180058)
	github.com/hashicorp/nomad/main.go:147 +0x55c
main.Run(...)
	github.com/hashicorp/nomad/main.go:87
main.main()
	github.com/hashicorp/nomad/main.go:83 +0x65
```

Do I need to do something else?

Reply:
Also I do not think using a pointer would work:
```
package main

import (
	"flag"
	"fmt"
)

func main() {
	testFlag := true
	flag.BoolVar(&testFlag, "flagname", false, "help message for flagname")
	fmt.Println(testFlag)
}
```
this would set `testFlag` to false, independent on whether passed or not and I cannot simply pass `nil` as default either.

Reply:
I think it would need a custom flag helper; but having spent a little more time thinking about it I don't think we need to be able to tell the difference between a user setting false, and the default value. Lets therefore ignore the idea of using a pointer and focus on not mutating the default value and using the env var at a later stage.

Reply:
Ok, I'll be on holiday next week but after that I will see what I can do. How "expensive" are calls to the system env in go? `Colorize` is used quite often, even during plan and while it may not immediately be noticable I rather not have plan call the getenv system call 100 times to render the output.